securedrop_client/crypto.py

```diff
@@ -37,8 +39,39 @@ class CryptoError(Exception):
     pass
 
 
-class GpgHelper:
+def read_gzip_header_filename(filename: str) -> str:
+    """
+    Extract the original filename from the header of a gzipped file.
+
+    Adapted from Python's gzip._GzipReader._read_gzip_header.
+    """
+    original_filename = ""
+    with open(filename, "rb") as f:
+        magic = f.read(2)
+        if magic != b"\037\213":
+            raise OSError("Not a gzipped file (%r)" % magic)
```

Comment:
just noting that this will get caught in the api base class and will emit the failure signal and the queue will move onto the next download file job